internal/tools/weather/http.go

```
@@ -0,0 +1,393 @@
+		}
+		body, readErr := readLimited(resp.Body, maxResponseBytes)
+		closeErr := resp.Body.Close()
+		if readErr != nil {
+			last = fmt.Errorf("%w: weather provider response is too large or unreadable", contracts.ErrDataIncomplete)
+			if attempt == c.maxRetries {
+				return last
+			}
+			continue
+		}
+		if closeErr != nil {
+			last = fmt.Errorf("close weather response: %w", closeErr)
```

golangci-lint (2.12.2)

[error] ineffectual assignment to last

(ineffassign)